src/offset.js

```diff
@@ -93,9 +94,26 @@ jQuery.fn.extend({
 
 		docElem = doc.documentElement;
 
-		// Make sure it's not a disconnected DOM node
-		if ( !jQuery.contains( docElem, elem ) ) {
-			return box;
+		// We have element in ShadowDOM, need recursive traversal from element to its parent
+		if ( elem.matches && elem.matches( ":host *" ) ) {
+			if ( !elem.getBoundingClientRect ) {
```

Comment:
Is there any situation where a browser would support shadowDOM but not `getBoundingClientRect`? seems unlikely.